src/frontend/src/lib/types/dapp.ts

```diff
@@ -0,0 +1,32 @@
+import { z } from 'zod';
+
+// see https://github.com/dfinity/portal/tree/95c67a5cfe201e4e5cb79f3cf5d18fe16498cd8c?tab=readme-ov-file#object-schema
+const dAppSchema = z.object({
```

Comment:
Thanks. As I mentionned in my very first comment on the review, it would be nice to provide the type / zod object in a separate PR for separation of converns. I can review it here but for future PR, please try to split logic and presentation.